core/src/test/java/io/confluent/rest/MetricsListenerIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+    TestMetricsReporter.reset();
+    Properties props = new Properties();
+    props.put(RestConfig.LISTENERS_CONFIG, "app1://localhost:" + getFreePort() + ","
+        + "app2://localhost:" + getFreePort());
+    props.put(RestConfig.LISTENER_PROTOCOL_MAP_CONFIG, "app1:http,app2:http");
+    TestRestConfig config = new TestRestConfig(props);
+    server = new ApplicationServer<RestConfig>(config);
+    TestApplication app1 = createApp("app1");
+    ((ApplicationServer<?>) server).registerApplication(createApp("app2"));
+    ((ApplicationServer<?>) server).registerApplication(app1);
+    server.start();
+
+    for (Connector connector : server.getConnectors()) {
+      if (connector.getName().equals("app1")) {
+        app1Endpoint =
+            "http://localhost:" + ((NetworkTrafficServerConnector) connector).getLocalPort();
+      } else if (connector.getName().equals("app2")) {
+        app2Endpoint =
+            "http://localhost:" + ((NetworkTrafficServerConnector) connector).getLocalPort();
+      }
+    }
+    client = ClientBuilder.newClient(app1.resourceConfig.getConfiguration());
+  }
+
+  private TestApplication createApp(String name) {
+    Properties props = new Properties();
+    props.put(RestConfig.METRICS_REPORTER_CLASSES_CONFIG, TestMetricsReporter.class.getName());
+    props.put(RestConfig.METRICS_JMX_PREFIX_CONFIG, "rest-utils." + name);
+    TestRestConfig config = new TestRestConfig(props);
+    return new TestApplication(config, name);
+  }
+
+  @AfterEach
+  public void tearDown() throws Exception {
+    server.stop();
+    server.join();
+    client.close();
+  }
+
+  @Test
+  public void testMetricsListenerRegisteredCorrectly() {
+    // send 5 requests to app1
+    for (int i = 0; i < 5; i++) {
+      Response response = client.target(app1Endpoint)
+          .path("/public/hello")
+          .request(MediaType.APPLICATION_JSON_TYPE)
+          .get();
+      assertEquals(Status.OK.getStatusCode(), response.getStatus());
+    }
+    // send 1 request to app2
+    for (int i = 0; i < 1; i++) {
+      Response response = client.target(app2Endpoint)
+          .path("/public/hello")
+          .request(MediaType.APPLICATION_JSON_TYPE)
+          .get();
+      assertEquals(Status.OK.getStatusCode(), response.getStatus());
+    }
+
+    for (KafkaMetric metric : TestMetricsReporter.getMetricTimeseries()) {
+      if (metric.metricName().group().equals("jetty-metrics")
+          && metric.metricName().name().equals("connections-active")
+      ) {
+        if (metric.metricName().tags().getOrDefault("instance-name", "").equals("app1")) {
+          // 5 connections-active in app1
+          assertConnectionsActiveMetrics(metric, /*expected value*/ 5.0);
+        } else if (metric.metricName().tags().getOrDefault("instance-name", "")
+            .equals("app2")) {
+          // 1 connections-active in app2
+          assertConnectionsActiveMetrics(metric, /*expected value*/ 1.0);
+        }
```

Comment:
can you check that else {} condition is never hit in the test? maybe throw, or assert.

Reply:
added throw for not reachable else path